### PR TITLE
Contrib/junit5

### PR DESCRIPTION
**Pull request by uguy**
Junit5 support for the Java option. Adding mockito and AssertJ as default mock and assertion libs.

**Warning** : This branch is based on the `contrib/no-parent` branch because some plugin in the old parent pom conflicts with Junit5 configuration.

### CODE REVIEW

File: src/main/resources/archetype-resources/src/main/resources-filtered/__artifactId__.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Custom.category=Custom connectors
-connectorDefinitionDescription=
-connectorDefinitionLabel=${artifactId}
+connectorDefinitionDescription=Retrieve data about a star wars character
```
Comment (rbioteau) [inline src/main/resources/archetype-resources/src/main/resources-filtered/__artifactId__.properties:2]: Fix descrption

File: src/test/resources/projects/testGroovyProject/reference/src/main/resources-filtered/connector-groovy-test.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Custom.category=Custom connectors
-connectorDefinitionDescription=
-connectorDefinitionLabel=connector-groovy-test
+connectorDefinitionDescription=Retrieve data about a star wars character
```
Comment (rbioteau) [inline src/test/resources/projects/testGroovyProject/reference/src/main/resources-filtered/connector-groovy-test.properties:2]: Fix description

File: src/test/resources/projects/testKotlinProject/reference/pom.xml [hunk 3]
```diff
@@ -1,105 +1,200 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.platform</groupId>
+            <artifactId>junit-platform-runner</artifactId>
+            <version>${junit-platform-runner.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.assertj</groupId>
```
Comment (rbioteau) [inline src/test/resources/projects/testKotlinProject/reference/pom.xml:83]: Is Assertj not a too opinionated choice ?

File: src/test/resources/projects/testKotlinProject/reference/src/main/resources-filtered/connector-kotlin-test.properties [hunk 4]
```diff
@@ -1,6 +1,6 @@
 Custom.category=Custom connectors
-connectorDefinitionDescription=
-connectorDefinitionLabel=connector-kotlin-test
+connectorDefinitionDescription=Retrieve data about a star wars character
```
Comment (rbioteau) [inline src/test/resources/projects/testKotlinProject/reference/src/main/resources-filtered/connector-kotlin-test.properties:2]: fix description

File: src/main/resources/archetype-resources/groovy-pom.xml [hunk 5]
```diff
@@ -1,120 +1,165 @@
...
+
+        <!-- Connector -->
+        <connector-definition-id>${project.artifactId}</connector-definition-id>
+        <connector-definition-version>1.0</connector-definition-version>
+        <connector-impl-id>${connector-definition-id}-impl</connector-impl-id>
+        <connector-impl-version>${project.version}</connector-impl-version>
+        <connector-main-class>${package}.${className}</connector-main-class>
+
+        <!-- Bonita -->
+        <bonita.engine.version>7.10.0</bonita.engine.version>
```
Comment (rbioteau) [inline src/main/resources/archetype-resources/groovy-pom.xml:26]: why is it fixed ?

File: src/test/resources/projects/testKotlinProject/reference/pom.xml (hunk 3 above)
Comment (uguy) [reply src/test/resources/projects/testKotlinProject/reference/pom.xml:83]: To be honest, AssertJ is the best assert lib for javaI have used, just like Mockito for mocking.
I find it easier to remove or replace two dependencies than writing all the boiler plate xml to have them. It also engage the user to write tests and readable ones :)

- groovy: spock
- java: junit5, assertj, mockito
- kotlin: junit5, assertj, mockk

File: src/main/resources/archetype-resources/kotlin-pom.xml [hunk 6]
```diff
@@ -1,105 +1,200 @@
...
-			<version>${kotlin.version}</version>
-		</dependency>
-		<dependency>
-			<artifactId>junit</artifactId>
-			<groupId>junit</groupId>
-			<scope>test</scope>
-		</dependency>
-	</dependencies>
+        <!-- Bonita -->
+        <bonita.engine.version>7.10.0</bonita.engine.version>
```
Comment (rbioteau) [inline src/main/resources/archetype-resources/kotlin-pom.xml:28]: Why is it fixed ?

File: src/test/resources/projects/testGroovyProject/reference/src/main/resources-filtered/connector-groovy-test.properties (hunk 2 above)
Comment (uguy) [reply src/test/resources/projects/testGroovyProject/reference/src/main/resources-filtered/connector-groovy-test.properties]: fixed

File: src/main/resources/archetype-resources/kotlin-pom.xml (hunk 6 above)
Comment (uguy) [reply src/main/resources/archetype-resources/kotlin-pom.xml:28]: my bad !

Comment (uguy) [reply src/main/resources/archetype-resources/kotlin-pom.xml]: fix pushed

File: src/main/resources/archetype-resources/groovy-pom.xml (hunk 5 above)
Comment (uguy) [reply src/main/resources/archetype-resources/groovy-pom.xml]: fix pushed

**Comment by rbioteau** [comment]
⚠️ to be applied on https://github.com/bonitasoft/bonita-actorfilter-archetype

File: src/test/resources/projects/testKotlinProject/reference/pom.xml (hunk 3 above)
Comment (rbioteau) [reply src/test/resources/projects/testKotlinProject/reference/pom.xml:84]: I checked and it seems that it has become pretty standard. 
So ok for me.

**Comment by uguy** [comment]
Closed: PR was moved to the branch contrib/no-parent

**Pull request closed without merging**